cogs/displayinfo.py

```diff
@@ -0,0 +1,116 @@
+import discord
+import constantvariables
+import mongoDB
+from discord.ext import commands
+from discord.ext import tasks
+from datetime import datetime
+from datetime import date
+import datetime as dt
+from pymongo.mongo_client import MongoClient
+from pymongo.server_api import ServerApi
+import asyncio
+
+
+
+
+# Create a new client and connect to the MongoDB server
+mongoClient = MongoClient(mongoDB.MONGODB_URI, server_api=ServerApi('1'))
+
+
+try:
+    # Send a ping to confirm a successful connection to MongoDB
+    mongoClient.admin.command('ping')
+    print("Pinged your deployment. You successfully connected to MongoDB!")
+except Exception as e:
+    print(e)
+
+# Select the database
+db = mongoClient.Rasenbot
+
+
+
+class DisplayInfo(commands.Cog):
+
+    def __init__(self,client):
+        self.client = client
+        self.timers = {}
+        self.current_collection = ""
+        self.current_guild_name = ""
+        self.count = 0
+        self.current_date = str(date.today())
+
+    # The 'mem_before' parameter represents the member's previous state, and 'mem_after' represents the current state.
+    @commands.Cog.listener()
+    async def on_member_update(self, mem_before:discord.Member, mem_after:discord.Member):
```

Comment:
Should add an initial check here to break `if mem_before.activity and mem_before.activity == mem_after.activity`. This would mean that activity is not `None` and they are the same activity. `on_member_update` could have been triggered because of a status update or something but we don't need to retrieve the database in that case.

Reply:
The one case we need to cover is if while someone is playing a game, they go offline. So we can check for that singular case and just add their timer until then.